R/learner_xgboost_surv_xgboost.R

```diff
@@ -151,43 +161,65 @@ delayedAssign(
     ),
 
     private = list(
-      .train = function(task) {
-
-        pv = self$param_set$get_values(tags = "train")
-
-        if (is.null(pv$objective)) {
-          pv$objective = "survival:cox"
-        }
-
-        data = task$data(cols = task$feature_names)
-        target = task$data(cols = task$target_names)
+      # helper function to construct an `xgb.DMatrix` object
+      .get_data = function(task, pv, row_ids) {
+        # use all task rows if `rows_ids` is not specified
+        if (missing(row_ids))
```

Comment:
don't use `missing()`, set default to `NULL` and use `if (is.null(row_ids))`

Reply:
Although it is not so important here, this is generally a good practice. Otherwise it is not understandable from the function signature that the function has a default value.

Reply:
Thanks for the tip!